packages/frontend/helpers/chains.ts

```diff
@@ -1,139 +1,84 @@
-  },
-  {
-    id: "0x89",
-    token: "MATIC",
-    label: "Polygon",
-    rpcUrl: "https://matic-mainnet.chainstacklabs.com",
-    blockExplorerUrl: "https://polygonscan.com/",
-    icon: "Polygon",
-  },
-  {
-    id: "0x64",
-    token: "xDai",
-    label: "Gnosis",
-    rpcUrl: "https://rpc.gnosischain.com/",
-    blockExplorerUrl: "https://gnosisscan.io/",
-    icon: "Gnosis",
-  },
-  {
-    id: "0xa4b1",
-    token: "AETH",
-    label: "Arbitrum",
-    rpcUrl: "https://arb1.arbitrum.io/rpc",
-    blockExplorerUrl: "https://arbiscan.io/",
-    icon: "Arbitrum",
-  },
-  {
-    id: "0xa",
-    token: "ETH",
-    label: "Optimism",
-    rpcUrl: "https://optimism-mainnet.public.blastapi.io/",
-    blockExplorerUrl: "https://optimistic.etherscan.io/",
-    icon: "Optimism",
-  },
-]
-export const testChains: Chain[] = [
-  {
-    id: "0x13881",
-    token: "MATIC",
-    label: "Mumbai",
-    rpcUrl: "https://matic-mainnet.chainstacklabs.com",
-    blockExplorerUrl: "https://mumbai.polygonscan.com/",
-    icon: "Polygon",
-  },
-  {
-    id: "0x5",
-    token: "GTH",
-    label: "Goerli",
-    rpcUrl: `https://goerli.infura.io/v3/${process.env.NEXT_PUBLIC_INFURA_KEY}`,
-    blockExplorerUrl: "https://goerli.etherscan.io/",
-    icon: "Ethereum",
-  },
-  {
-    id: "0x1a4",
-    token: "ETH",
-    label: "Optimism Goerli",
-    rpcUrl: "https://goerli.optimism.io/",
-    blockExplorerUrl: "https://goerli-optimism.etherscan.io/",
-    icon: "Optimism",
-  },
+export const testChains = [
+  CHAIN[ChainId.GOERLI],
+  CHAIN[ChainId.MATIC_MUMBAI],
+  CHAIN[ChainId.OPTIMISM_GOERLI],
 ]
 
-if (process.env.NEXT_PUBLIC_APP_ENV === "development") {
-  chains.push(...testChains)
-}
+// if (process.env.NEXT_PUBLIC_APP_ENV === "development") {
+//   chains.push(...testChains)
+// }
 
-const chainsMap = new Map<string | number, string>()
-const explorersMap = new Map()
-chains.map((c) => {
-  chainsMap.set(c.id, c.label) // string hex id
-  chainsMap.set(parseInt(c.id), c.label) // num id
+const rpcs = {
+  [ChainId.ETHEREUM]: `https://mainnet.infura.io/v3/${process.env.NEXT_PUBLIC_INFURA_KEY}`,
+  [ChainId.MATIC]: "https://matic-mainnet.chainstacklabs.com",
+  [ChainId.FANTOM]: "https://rpcapi.fantom.network",
+  [ChainId.ARBITRUM]: "https://arb1.arbitrum.io/rpc",
+  [ChainId.OPTIMISM]: "https://optimism-mainnet.public.blastapi.io/",
+  [ChainId.GNOSIS]: "https://rpc.gnosischain.com/",
+  [ChainId.GOERLI]: `https://goerli.infura.io/v3/${process.env.NEXT_PUBLIC_INFURA_KEY}`,
+  [ChainId.MATIC_MUMBAI]: "https://matic-mainnet.chainstacklabs.com",
+  [ChainId.OPTIMISM_GOERLI]: "https://goerli.optimism.io/",
+}
 
-  explorersMap.set(c.id, c.blockExplorerUrl)
-  explorersMap.set(parseInt(c.id), c.blockExplorerUrl)
+export const onboardChains: OnboardChain[] = chains.map((c) => {
+  return {
+    id: chainIdToHex(c.chainId),
+    token: c.nativeTokenName,
+    label: c.label,
+    blockExplorerUrl: c.blockExplorerUrl,
+    rpcUrl: rpcs[c.chainId],
```

Comment:
The rpc url is already available here:
```
rpcUrl: c.connection.rpcProvider.connection.url
```
so no need of the const `rpcs`

Reply:
`connection` can be `undefined`, I was thinking of using `??` but not sure what can we use as value without using the `rpcs` const.

Reply:
I see, I think it's safe to assume the connection is not `undefined` because we set the chain connection while initializing the sdk.
Maybe you can use the rpcs defined here as fallbacks. So when new chains get added and we forgot to update here, everything would work fine.